get-started/commit-code-and-submit-a-pull-request.md

```diff
@@ -1,11 +1,104 @@
 # Commit code and submit a pull request
 
-including but not limited to:
+The TiDB project uses [Git](https://git-scm.com/) to manage its source code. To contribute to the project, you need to get familiar with Git features so that your changes can be incorporated into the codebase.
 
-* how to organize branches in your local git repository
-* what format needs to be followed for commit message
-* what format needs to be followed for pull request title
-* what content needs to be filled in the pull request description
-* how to request pull request review from reviewers
-* how to get notified when the pull request received review comments
+The goal of this page is to cover some of the most common questions and problems that new contributors might face. Although this page covers some Git basics too, if you find that the content on this page is a little difficult to understand, it is recommended that you read some introductions to Git first as follows: 
 
+- Beginner and Getting started sections of [this tutorial] (https://www.atlassian.com/git/tutorials) from Atlassian 
+- [Documentation](https://docs.github.com/en/github/getting-started-with-github/set-up-git) and [guides](https://guides.github.com/introduction/git-handbook/) for beginners from Github
+- More in-depth [book](https://git-scm.com/book/en/v2/) from Git
+
+## Prerequisites
+
+Before creating a pull request, you need to make sure that you've installed Git, forked [pingcap/tidb](https://github.com/pingcap/tidb), and cloned the upstream repo to your PC. The instructions on this page use the command line interface to interact with Git; there are also several GUIs and IDE integrations that can interact with Git too.
+
+If you've cloned the upstream repo, you will be able to reference it using `origin` in your local repo. Next, you need to set up a remote for the repo your forked using the following command:
+
+```bash
+git remote add dev https://github.com/username/tidb.git
+```
+
+You can check the remote setting using the following command:
+
+```bash
+git remote -v
+# dev    https://github.com/username/tidb.git (fetch)
+# dev    https://github.com/username/tidb.git (push)
+# origin    https://github.com/pingcap/tidb.git (fetch)
+# origin    https://github.com/pingcap/tidb.git (push)
+```
+
+## Standard Process
+
+The following is a normal procedure that you're likely to use for the most common minor changes and PRs:
+
+1. Ensure that you're making your changes on top of master: `git checkout master`.
```

Comment:
How about this:

1. Ensure that you're making your changes on top of the latest master branch of origin ([pingcap/tidb](https://github.com/pingcap/tidb.git)):
  ```sh
  git checkout master
  git pull
  ```

2. Create a new branch for your changes:
  ```sh
  git checkout -b fix-${issue_number}
  ```

3. Make some changes in your local branch and test them to ensure it works as expected.

4. Commit your changes and push them to your `dev` remote repository:
  ```sh
  # stage the files you created/changed/deleted
  git add xx

  # commit the staged changes, make sure the commit message is meaningful and readable
  git commit -m "what's changed"

  # push the local change to your forked remote repository
  git push --set-upstream dev fix-${issue_number}
  ```

5. Create a pull request (PR, for short) from your fork to the `master` branch of pingcap/tidb. For more information on how to make a PR, see [Making a Pull Request](https://guides.github.com/activities/forking/#making-a-pull-request) in GitHub Guides. Typically, the URL to create a PR based on your fork is:
  ```text
  https://github.com/${your_github_id}/tidb/pull/new/{your_branch_name}
  ```

Reply:
I'd like to remain as is. The change set seems nearly the same.

Reply:
Yes, the content is almost the same. There are “fewer” steps compared with the original one:

- fewer words in each step, makes the audience able to grab the key points
- make the commands wrapped in code blocks instead of in the sentences, make the original sentences more readable, and the code is also copyable by the audience.